### PR TITLE
Add embedded config to analytics payload

### PR DESCRIPTION
**Pull request by wooj-stripe**
## Summary
Adds configuration for EmbeddedPaymentElement.Configuration to Analytics payload

## Motivation
Need to track

## Testing
manual testing

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetAppearance.swift [hunk 1]
```diff
@@ -306,5 +308,18 @@ public extension PaymentSheet {
...
+            var payload = [String: Any]()
+            payload["style"] = style != PaymentSheet.Appearance.EmbeddedPaymentElement.default.style
+            payload["row"] = row != PaymentSheet.Appearance.EmbeddedPaymentElement.default.row
+            // Convenience payload item to make querying high level appearance usage easier
+            payload["usage"] = payload.values.contains(where: { value in
+                if let boolValue = value as? Bool {
+                    return boolValue == true
+                }
+                return false
+            })
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetAppearance.swift:321]: Do we need this?

Comment (wooj-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetAppearance.swift:321]: I thought it was a common pattern we were using -- happy to remove if we don't want it.

Comment (porter-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetAppearance.swift:321]: I think the only place we use this is for top level appearance stuff so we can easily answer "Is appearance being used at all". Peaceful to remove this

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetAppearance.swift [hunk 2]
```diff
@@ -306,5 +308,11 @@ public extension PaymentSheet {
                 public var spacing: CGFloat = 12.0
             }
         }
+        var analyticPayload: [String: Any] {
```
Comment (yuki-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetAppearance.swift:311]: Nit but wdyt about defining this and the `static let` above in an extension in `STPAnalyticsClient+PaymentSheet`?

Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetAppearance.swift:311]: @porter-stripe Outside scope of this PR  and embedded too really but I wonder if it'd be more useful to send the actual value of these things.

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElementConfiguration.swift [hunk 3]
```diff
@@ -141,6 +141,15 @@ extension EmbeddedPaymentElement {
...
             case `continue`
+
+            var analyticValue: String {
+                switch self {
+                case .confirm:
+                    return "confirm"
+                case .`continue`:
+                    return "continue"
+                }
+            }
```
Comment (yuki-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElementConfiguration.swift:152]: Ah sorry, could we move this to `STPAnalyticsClient+PaymentSheet` too?

Comment (wooj-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElementConfiguration.swift]: oops, my bad, thanks!

**Review by yuki-stripe — APPROVED** [review]

**Pull request merged**